proposals/0387-cross-compilation-destinations.md

```diff
@@ -80,85 +86,93 @@ The proposed structure of artifact bundles containing CC destinations looks like
+│ ├ aarch64-unknown-linux-gnu
 │ │ ├ toolset.json
-│ │ └ <destination file tree>
-│ └ x86_64-apple-darwin
-│   ├ destination.json
+│ │ └ <triple-specific destination files and directories>
+│ └ x86_64-unknown-linux-gnu
 │   ├ toolset.json
-│   └ <destination file tree>
+│   └ <triple-specific destination files and directories>
 ├ ubuntu_focal
-│ └ x86_64-apple-darwin
-│   ├ destination.json
+│ ├ destination.json
+│ └ x86_64-unknown-linux-gnu
 │   ├ toolset.json
-│   └ <destination file tree>
+│   └ <triple-specific destination files and directories>
 ├ ubuntu_bionic
 ┆ └┄
 ```
 
-Here each artifact directory is dedicated to a specific CC destination, while binaries for a specific host platform are
-placed in `arm64-apple-darwin` and `x86_64-apple-darwin` subdirectories.
-
-`info.json` bundle manifests at the root of artifact bundles should specify `"type": "crossCompilationDestination"`
-for corresponding artifacts. Artifact identifiers in this manifest uniquely identify a CC destination. The rest of the
-properties of bundle manifests introduced in SE-0305 are preserved.
+Here each artifact directory is dedicated to a specific CC destination, while files specific to each triple are placed
+in `aarch64-unknown-linux-gnu` and `x86_64-unknown-linux-gnu` subdirectories.
 
-### `destination.json` files
+`info.json` bundle manifests at the root of artifact bundles should specify `"type": "crossCompilationDestination"` for
+corresponding artifacts. Artifact identifiers in this manifest file uniquely identify a CC destination, and
+`supportedTriples` property in `info.json` should contain build-time triples that a given destination supports. The rest
+of the properties of bundle manifests introduced in SE-0305 are preserved.
 
-Note the presence of `destination.json` files in each `<host variant>` subdirectory. These files should contain a JSON
-dictionary with an evolved version of the schema of [existing `destination.json` files that SwiftPM already
-supports](https://github.com/apple/swift-package-manager/pull/1098) and `destination.json` files presented in the pitch
-version of this proposal, hence `"schemaVersion": "3.0"`:
+Here's how `info.json` file could look like for `swift-5.8_ubuntu.artifactbundle` introduced in the example
+above:
 
-```json
+```json5
 {
-  "schemaVersion": "3.0",
-  "buildTimeTriples": ["<an array of supported build-time platform triples>"],
-  "runTimeTriples": ["<an array of supported run-time platform triples>"],
-  "swiftResourcesPaths": ["<array with relative paths containing Swift resources in the destination tree>"],
-  "includeSearchPaths": ["<array with relative paths containing headers in the destination tree>"],
-  "librarySearchPaths": ["<array with relative paths containing libraries in the destination tree>"],
+  "artifacts" : {
+    "swift-5.8_ubuntu22.04" : {
+      "type" : "crossCompilationDestination",
+      "version" : "0.0.1",
+      "toolset": "toolset.json",
```

Comment:
As far as I read, the `"toolset": "toolset.json"` entry is not mentioned anywhere. Could you clarify the semantics of this field?

Reply:
Sorry for the confusion, that's vestigial from previous drafts where I considered specifying toolsets in `info.json`. That's cleaned up now.